blockchain.go

```diff
@@ -675,44 +675,61 @@ func (b *Blockchain) GetAccount(address sdk.Address) (*sdk.Account, error) {
 	b.mu.RLock()
 	defer b.mu.RUnlock()
 
-	flowAddress := sdkconvert.SDKAddressToFlow(address)
-
-	account, err := b.getAccount(flowAddress)
+	latestBlock, err := b.GetLatestBlock()
 	if err != nil {
 		return nil, err
 	}
 
-	sdkAccount, err := sdkconvert.FlowAccountToSDK(*account)
+	return b.GetAccountAtBlock(address, latestBlock.Header.Height)
```

Comment:
`GetAccount` locks and then `GetAccountAtBlock` locks again. See https://pkg.go.dev/sync#RWMutex.RLock.


Maybe replace with `getAccountAtBlock`, or, given that the body after the locking part at the beginning of this function is then equivalent to `getAccount`, just call `getAccount`.

Reply:
Oh right, thank you, nice catch.